fixed_bed_model.py

```diff
@@ -0,0 +1,2630 @@
+                    elif alpha_option == 4: 
+                        alpha_ = 1.0
+
+                    if alpha_variable:
+                        m.alpha[j,'CO2',z,i] = alpha_
+
+                if m.chemsorb:
+                    m.nchemstar[j,'CO2',z,i] = Nchem_e(z_, t_)[0]
+                    m.dnchemdt[j,'CO2',z,i] = dndt_chem(z_, t_)[0]
+                    m.nchem[j,'CO2',z,i] = Nchem(z_, t_)[0]
+                    if m.energy:
+                        m.nchemstar_mod[j,'CO2',z,i] = alpha_*value(m.nplin[j,z,i])*solid_pres(z_, t_)[0]/plin + (1-alpha_)*Nchem_e(z_, t_)[0]
+                    else:
+                        m.nchemstar_mod[j,'CO2',z,i] = alpha_*m.nplin*solid_pres(z_, t_)[0]/plin + (1-alpha_)*Nchem_e(z_, t_)[0]
+
+                if m.physsorb:
+                    m.nphysstar[j,'CO2',z,i] = Nphys_e(z_, t_)[0]
+                    #m.nphysstar_mod[j,'CO2',z,i] = alpha_*m.nplin*solid_pres(z_, t_)[0]/plin + (1-alpha_)*Nphys_e(z_, t_)[0]
+                    m.dnphysdt[j,'CO2',z,i] = dndt_phys(z_, t_)[0]
+                    m.nphys[j,'CO2',z,i] = Nphys(z_, t_)[0]   
+            
+            
+
+
+def compute_Kp(mod,LOUD=True):
+    ''' Compute Kp from the solution [m^2/s]
+    '''
+    Kp_ = {}
+    for c in mod.SCOMPS:
+        tfc = value(mod.fitted_transport_coefficient[c])
+
+        Kp_[c] = radp*radp/15/ads_epsp/tfc
+        
+        if LOUD:
+            print("fitted_transport_coefficient[",c,"] =",tfc,"s")
+            print("Kp[",c,"] =",Kp_[c],"m^2/s")
+        
+    return Kp_
+
+
+### DoE analysis function
+def ext_pres_double(total, option1, option2, node):
+    '''
+    Extract the solution of a variable
+    Arguments:
+        total: a pandas dataframe of solutions
+        option: name of the variable
+        node: number of timesteps
+    Note: the Ngrid is fixed to 20 
+    Return: a list
+    '''
+    x_py1 = total[option1]
+    x_pyomo1 = np.zeros((20,node))
+
+    for i in range(0,20):
+        for j in range(0,node):
+            x_pyomo1[i,j] = x_py1.iloc[i*node+j]
+            
+    x_py2 = total[option2] 
+    x_pyomo2 = np.zeros((20,node))
+
+    for m in range(0,20):
+        for k in range(0,node):
+            x_pyomo2[m,k] = x_py2.iloc[m*node+k]        
+        
+    return x_pyomo1, x_pyomo2
+
+def average(lis):
+    '''
+    Get the average value of velocity/pressure etc. 
+    '''
+    li = np.asarray(lis)
+    res = li.reshape([69*20,])
+    return sum(res)/len(res)
+
+def time_points(n, time_range):
+    '''
+    Split the timespan of the experiment into n nodes
+    
+    Argument: n: the number of time nodes.
+    
+    return: time_points, a list of values of timepoints in [s]
+    '''
+    time_points = []
+    int_time = time_range/n
+    
+    # specify where the time begins. Usually 0
+    time_ele = 0
+    time_points.append(time_ele)
+    for i in range(n):
+        time_ele += int_time
+        time_points.append(time_ele)
+
+    return time_points
+
+def Residual(m):
```

Comment:
Is there a reason this is capitalized? Capitalized names imply classes in Python (per convention).